File: 백준 문제 풀이 모음/class 3/11659.py
import sys
N, M = map(int, sys.stdin.readline().split())
num = list(map(int, sys.stdin.readline().rstrip().split()))

for x in range(1, len(num)) :
    num[x] += num[x-1]


for _ in range(M) :
    i, j = map(int, sys.stdin.readline().split())
    if i == 1 :
        count = num[j-1] 
    elif i == j :
        count = num[j-1] - num[j-2]
    else :
        count = num[j-1] - num[i-2]

    print(count)

File: 백준 문제 풀이 모음/kotlin/class 3/11659.kt
import java.util.StringTokenizer

fun main() = with(System.`in`.bufferedReader()){
    val str = readLine().split(' ').toList()
    val n = str[0].toInt()
    val m = str[1].toInt()

    val list = mutableListOf<Long>()
    var count : Long = 0
    val str2 = StringTokenizer(readLine())
    repeat(n) {
        count += str2.nextToken().toLong()
        list.add(count)
    }
//    println(list)


    val sb = StringBuilder()

    repeat(m) {
        val input = readLine().split(' ')
        val a = input[0].toInt() - 1
        val b = input[1].toInt() - 1

        if (a == 0) {
            sb.append(list[b])
            sb.append('\n')
        } else {
            sb.append(list[b] - list[a-1])
            sb.append('\n')
        }

    }

    println(sb.toString())




}